visidata/loaders/archive.py

```diff
@@ -0,0 +1,89 @@
+import codecs
+import tarfile
+import zipfile
+
+from visidata import *
+
+
+class ArchiveSheet(Sheet):
+    'Provide wrapper around `zipfile` and `tarfile` libraries for opening archive files.'
+    rowtype = 'files' # rowdef ZipInfo or TarInfo
+
+    def __init__(self, p, archive_type='zip'):
+        super().__init__(p.name, source=p)
+        self.archive_type = archive_type
+
+    @asyncthread
+    def reload(self):
+
+        if self.archive_type == 'zip':
+            self.columns = [
+                ColumnAttr('filename'),
+                ColumnAttr('file_size', type=int),
+                Column('date_time', type=date,
+                       getter=lambda col, row: datetime.datetime(*row.date_time)),
+                ColumnAttr('compress_size', type=int)
+            ]
+        if self.archive_type == 'tar':
+            self.columns = [
+                ColumnAttr('name'),
+                ColumnAttr('size', type=int),
+                ColumnAttr('mtime', type=date),
+                ColumnAttr('type', type=int),
+                ColumnAttr('mode', type=int),
+                ColumnAttr('uname'),
+                ColumnAttr('gname')
+            ]
+
+        self.recalc()
+        if self.archive_type == 'zip':
+            contents = zipfile.ZipFile(self.source.resolve(), 'r').infolist()
+
+        if self.archive_type == 'tar':
+            contents = tarfile.open(name=self.source.resolve()).getmembers()
+
+        with Progress(total=len(contents)) as prog:
```

Comment:
Heh, this is actually a lot simpler:
```
for row in Progress(contents):
    self.addRow(row)
```

Everything else is handled for you.  The csv loader is more complicated because the number of rows isn't known beforehand, and the number of header rows can be variable.  Here, though, the `contents` has a proper `len` and there is no header row (we specify the columns manually).